elide-core/src/main/java/com/yahoo/elide/graphql/PersistentResourceFetcher.java

```diff
@@ -79,6 +117,151 @@ private void logContext(RelationshipOp operation, Environment environment) {
+     * @return list of persistent resource objects
+     */
+    private Object fetchObjects(Environment context) {
+        /* sanity check for data argument w FETCH */
+        if(context.data.isPresent()) {
+            throw new BadRequestException("FETCH must not include data");
+        }
+
+        /* check whether current object has a parent or not */
+        if (context.isRoot()) {
+            return fetchEntity(context);
+
+        } else {
+            EntityDictionary dictionary = context.requestScope.getDictionary();
+            Class parentClass = context.parentResource.getResourceClass();
+            String fieldName = context.field.getName();
+
+            if(dictionary.isAttribute(parentClass, fieldName)) { /* fetch attribute properties */
+                return fetchAttribute(context.parentResource, context.field);
+
+            } else if(dictionary.isRelation(parentClass, fieldName)){ /* fetch relationship properties */
+                return fetchRelationship(context);
+            } else {
+                throw new BadRequestException("Unrecognized object type: " + context.outputType.getClass().getName());
+            }
+        }
+    }
+
+    /**
+     * Returns the binding record type for a given entity
+     * @param requestScope request scope
+     * @param outputType GraphQL output type
+     * @return record type
+     */
+    private Class getRecordType(RequestScope requestScope, GraphQLType outputType) {
+        GraphQLObjectType graphQLType = (GraphQLObjectType) ((GraphQLList) outputType).getWrappedType();
+        String entityType = graphQLType.getName();
+        Class recordType = requestScope.getDictionary().getEntityClass(entityType);
+        if (recordType == null) {
+            throw new UnknownEntityException(entityType);
+        } else {
+            return recordType;
+        }
+    }
+
+    /**
+     * Fetches an entity, can be root or a nested entity
+     * @param context environment encapsulating graphQL's request environment
+     * @return persistent resource object(s)
+     */
+    private Object fetchEntity(Environment context) {
+        Class recordType = getRecordType(context.requestScope, context.outputType);
+        return fetchObject(context.requestScope, recordType, context.ids, context.outputType, context.sort,
+                context.offset, context.first, context.filters);
+    }
+
+    /**
+     * FETCH top level entity either by ids or as a collection
+     * @param requestScope request scope
+     * @param recordType record type
+     * @param ids list of ids (can be NULL)
+     * @param sort sort by ASC/DESC
+     * @param offset pagination offset argument
+     * @param first pagination first argument
+     * @param filters filter params
+     * @return list of persistent resource objects
+     */
+    private Object fetchObject(RequestScope requestScope, Class recordType, Optional<List<String>> ids,
```

Comment:
The naming of fetchEntity and fetchObject is inconsistent.  They are fetching the same thing - so I don't think we should name the function differently.